File: sgcmapi/src/main/java/br/ufac/sgcmapi/service/ICrudService.java
package br.ufac.sgcmapi.service;

import java.util.List;

public interface ICrudService<T> {

    public List<T> consultar(String termoBusca);
    public T consultar(Long id);
    public T salvar(T objeto);
    public void remover(Long id);
    
}

File: sgcmapp/src/app/service/i-crud-service.ts
import { Observable } from "rxjs";

export interface ICrudService<T> {
    apiUrl: string;
    consultar(termoBusca?: string): Observable<T[]>;
    consultarPorId(id: number): Observable<T>;
    salvar(objeto: T): Observable<number | void>;
    remover(id: number): Observable<void>;
}
